src/Waives.Http/LoggingRequestSender.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Waives.Http.Logging;
+
+namespace Waives.Http
+{
+    internal class LoggingRequestSender : IHttpRequestSender
+    {
+        private readonly IHttpRequestSender _wrappedRequestSender;
+        internal ILogger Logger;
```

Comment:
✋ I'm quite uneasy about this being exposed here, especially as it's a mutable field. What else references it?